File: 19-03-2025_Minimum_Operations_to_Make_Binary_Array_Elements_Equal_to_One_I.py
from typing import List

class Solution:
    def flip(self, i: int, nums: List[int]) -> None:
        for j in range(i, i + 3):
            nums[j] = 0 if nums[j] else 1
    
    def minOperations(self, nums: List[int]) -> int:
        zero_count = sum(1 for num in nums if num == 0)
        if not zero_count:
            return 0
        
        flips = 0
        n = len(nums)
        for i in range(n - 2):
            if nums[i] == 0:
                self.flip(i, nums)
                flips += 1
        
        zero_count = sum(1 for num in nums if num == 0)
        return flips if zero_count == 0 else -1

File: 19-03-2025_Minimum_Operations_to_Make_Binary_Array_Elements_Equal_to_One_I.java
public class Solution 
{
    private void flip(int i, int[] nums) 
    {
        for (int j = i; j < i + 3; j++) 
        {
            nums[j] ^= 1; 
        }
    }
    
    public int minOperations(int[] nums) 
    {
        int zeroCount = 0;
        for (int num : nums) 
        {
            if (num == 0) zeroCount++;
        }
        if (zeroCount == 0) return 0;
        
        int flips = 0;
        int n = nums.length;
        for (int i = 0; i < n - 2; i++) 
        {
            if (nums[i] == 0) 
            {
                flip(i, nums);
                flips++;
            }
        }
        

        zeroCount = 0;
        for (int num : nums) 
        {
            if (num == 0) zeroCount++;
        }
        return zeroCount == 0 ? flips : -1;
    }
}
